src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/util/WaitFor.java

```diff
@@ -16,16 +18,31 @@
      * @throws Exception If the time limit expires before the task evaluates to true
      */
     public static void waitFor(Callable<Boolean> task, long maxWait, TimeUnit unit) throws Exception {
+        waitFor(task, maxWait, unit, "waitFor timed out before task evaluated to true");
+    }
+
+    /**
+     * Waits at most the specified time for the given task to evaluate to true
+     * @param task The task which we hope evaluates to true before the time limit
+     * @param maxWait The max amount of time to wait for the task to evaluate for true
+     * @param unit The time unit of the maxWait parameter
+     * @param message The exception message to print if waitFor times out
+     * @throws Exception If the time limit expires before the task evaluates to true
+     */
+    public static void waitFor(Callable<Boolean> task, long maxWait, TimeUnit unit, String message)
+        throws Exception {
         long maxWaitMillis = TimeUnit.MILLISECONDS.convert(maxWait, unit);
-        long pollTime = System.currentTimeMillis();
-        long curTime;
+        Instant start = Instant.now();
         while (!task.call() && maxWaitMillis >= 0) {
-            curTime = System.currentTimeMillis();
-            maxWaitMillis -= (curTime - pollTime);
-            pollTime = curTime;
+            Instant finish = Instant.now();
+            maxWaitMillis -= Duration.between(start, finish).toMillis();
+            start = finish;
+            // Wait for a fixed amount before calling the task again
+            Thread.sleep(500L);
         }
-        if (maxWaitMillis < 0) {
-            throw new TimeoutException("WaitFor timed out before task evaluated to true");
+        // Check the task one last time before throwing an exception
+        if (!task.call() && maxWaitMillis < 0) {
```

Comment:
Maybe a `do-while` is a cleaner way to ensure we don't miss out on the last `500` millis ?

Reply:
Done